### PR TITLE
Including typing_extensions as dependency

### PR DESCRIPTION
**Pull request by gabrielcnr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by gabrielcnr** [comment]
I'm adding myself and @marcelotrevisani as maintainers here too.
I tried to add the selector `py<310` for the typing_extensions dependency but then it is not possible because the package is noarch. Since it's a very lightweight dependency and it's very likely to be in an environment because of other dependencies I'm suggesting we simply make that a dependency for the conda package.

**Comment by marcelotrevisani** [comment]
@conda-forge-admin, please rerender

**Comment by MatthieuDartiailh** [comment]
I would like to know if there is a standard policy within conda-forge for the case of noarch with a tiny version dependent dependency. 

If there is no consensus, I am personally fine adding it.

**Comment by marcelotrevisani** [comment]
> I would like to know if there is a standard policy within conda-forge for the case of noarch with a tiny version dependent dependency.
> 
> If there is no consensus, I am personally fine adding it.

Hi @MatthieuDartiailh the consensus is to not use the selectors when noarch for a few reasons, to not complicate stuff, to avoid any problems when mixing selectors with noarch and also the linter will complain. But in fact, the python selector can be used, just the platforms one and a few others that cannot. However, to avoid overcomplicating stuff, I would recommend to just letting that dependency in place as it is a small one, and in fact, the typing_extensions should be an empty package for python >=3.10 if I am not mistaken. It was discussed about that in a certain point but I do not remember anymore

File: recipe/meta.yaml
Comment (MatthieuDartiailh) [inline recipe/meta.yaml]: It may be worth double checking but I am not sure 3.6 is still supported. I think the oldest tesyed was 3.8.

**Review by MatthieuDartiailh — COMMENTED** [review]
LGTM modulo my concern with Python version.

**Comment by MatthieuDartiailh** [comment]
Thanks @marcelotrevisani I just wanted to be sure we were in line with other packages.

File: recipe/meta.yaml
Comment (gabrielcnr) [reply recipe/meta.yaml]: I'm making the change to pin/constrain >= 3.8

**Review by marcelotrevisani — APPROVED** [review]
I'm merging this as the maintainer agreed with the modifications

**Pull request merged**
